### ISSUE TITLE
SbBlokKeyDataTypes should support optional properties (undefined values)

### ISSUE THREAD

**Issue by Haprog** [storyblok/storyblok-js#133]
When adding new fields to existing custom blocks/components I want to be able to define optional properties (may be `undefined`) for my component types in TS (which are extending from `SbBlokData`) for backwards compatibility with component instances that are part of stories that have not been modified since editing the block definition since those old versions of the component don't yet contain the new property until they have been edited and saved.

Here's an example of what I want to do:
```ts
interface MyComponentBlok extends SbBlokData {
  some_field: string;
  new_field?: string; // <- currently TS doesn't allow this to be optional due to SbBlokKeyDataTypes
}
```

In the above case if I have just added the `new_field` to my block definition in Storyblok, the old instances of that block still don't have the property so I know I can't rely on it always being there yet (and I might have a lot of stories that don't contain the new field yet even if they use the block already) and I know I need to check for the existence of it in my code by TS will work against me if I can't declare it as optional.

In practice the API doesn't return those new properties for all cases but only for those cases that have been edited after introducing the new field so I want my TS types to reflect the reality that in some cases the property might not be there but in other cases it will so I would need to declare my new property as optional (e.g. `new_field?: string;`)

As a fix I suggest to update [this line](https://github.com/storyblok/storyblok-js/blob/5572b208333795cef5014d3a442c907e8da23e9b/lib/types.ts#L22):
```ts
export type SbBlokKeyDataTypes = string | number | object | boolean;
```
to:
```ts
export type SbBlokKeyDataTypes = string | number | object | boolean | undefined;
```

I can do a PR for this if you agree with this change.

**Comment by tricki** [comment]
I think this was fixed in PR #134, see issue https://github.com/storyblok/storyblok-js/issues/66

**Comment by Haprog** [comment]
Yes, thanks! I was looking at that issue just before you made the PR for it so I wasn't sure what it was about exactly, but that fix is exactly what I needed.
